vocabularies/UI.xml

```diff
@@ -1401,6 +1401,37 @@ The trend is
         <Annotation Term="Core.IsLanguageDependent" />
       </Term>
 
+      <Term Name="InputMask" Type="UI.InputMaskType" Nullable="false" AppliesTo="Property Parameter">
+        <Annotation Term="Core.Description" String="Properties or parameters annotated with this term will get a mask in edit mode." />
+        <Annotation Term="Core.LongDescription">
+          <String>Input masks not only improve readability, but also help to enter data correctly. 
+So, masks can be especially useful for input fields that have a fixed pattern, e.g. international bank account numbers or social security numbers. 
+[Here](../examples/UI.InputMask-sample.xml) you can find an example for this annotation</String>
+        </Annotation>
+      </Term>
+      <ComplexType Name="InputMaskType">
+        <Property Name="Mask" Type="Edm.String" Nullable="false">
+          <Annotation Term="Core.Description" String="The mask to be applied to the property" />
+        </Property>
+        <Property Name="Placeholder" Type="Edm.String" DefaultValue="_">
+          <Annotation Term="Core.Description" String="A single character symbol to be shown where the user can type a character" />
+        </Property>
+        <Property Name="InputMaskRule" Type="Collection(UI.InputMaskRuleType)" Nullable="false">
```

Comment:
How are these rules enforced?
* Is typing restricted to the allowed characters (immediate feedback)? Then users would need a visual indication which rule they have to obey at a given position (for example, different placeholders per rule).
* Or are rules checked only after the user has finished typing?

Reply:
Typing is restricted to allowed characters. A UI.Placeholder might help here :-) See: https://sapui5.hana.ondemand.com/#/entity/sap.m.MaskInput/sample/sap.m.sample.MaskInput